### ISSUE TITLE
How to implement division operation on ciphertext

### ISSUE THREAD

**Issue by wongwanying** [homenc/HElib#345]
**Detailed Description:**
It seems that Helib supports addition, multiplication and other operations. But if I want to encrypt floating-point numbers and divide between ciphertexts, what do I do?

**Comment by renaud-sirdey** [comment]
Hi, although you can in principe encode float-point numbers and perform any operations on them, it would be too costly to do so by means of FHE operators. Instead, I recommend to used fixed-point arithmetic i.e. you represent your floating point number by integers after premultiplying them by a suitable large (but not too large) constant. The Cingulata offers you right shifts (operator >>) which allows to perform divisions by powers of 2. So if you want to divide by a constant K you can  multiply by by say (2^16/K) and then right shift by 16. It's like in the old days when we had to program processors without FPUs... Dividing by a ciphertext value is more complicated and there is currently no operator to do that right now in Cingulata - from our experience however, it is not often REALLY needed in the kind of algorithm that are presently implemented in FHE (and within decent performance reach).
I hope this helps, Renaud

**Comment by JustKshitijD** [comment]
Yes, division between homomorphically encrypted ciphertext dividend and divisior seems like it can be done by polynomial division method. Has this not been implemented in HElib? Also, note that it is not developed in Palisade and other libraries as well.
